### DISCUSSION TITLE
AdNauseam adaptation to v3 manifest version

### DISCUSSION THREAD

**Discussion by mneunomne** [dhowe/AdNauseam#2188] [General]
In theory, AdNauseam on v3 manifest version should be based on uBlock Lite, which is the version of uBlock designed specifically for the v3 manifest. Documentation can be checked [here]( https://github.com/gorhill/uBlock/blob/master/platform/mv3/description/en.md), and discussion [here](https://github.com/uBlockOrigin/uBlock-issues/issues/338).

---------------

https://github.com/uBlockOrigin/uBlock-issues/issues/338#issuecomment-1253893421

This documentation is worth reading, the last part is somewhat concerning to us to understand if it is possible that we continue to be based on uBlock Origin or not for chrome users:

>Issues encountered when working on uBOL.

> The DNR's matching algorithm differs from uBO's matching algorithm regarding modifier filters (redirect=, csp=, removeparam), so currently all of these are dropped from uBOL. More time is needed to figure out how to best convert these filters despite the DNR's different matching algorithm compared to uBO's matching algorithm.

> Generic cosmetic filters are another issue to investigate and decide whether these will be supported or not.

> Given that cosmetic filtering is declarative, it's not possible to have an element picker to create cosmetic filters. This could be accomplish only through a service worker dynamically injecting those content filters, which as stated above would lead to point 2. -- something I want to avoid otherwise this would defeat the Lite in uBO Lite.

> Since all filtering is done declaratively, this means there is no way to force an update of filters when something is fixed in filter lists, the fix will have to be published in the next version of the extension uploaded to the CWS.

> Many users of uBO will dislike the limitations of uBOL when compared to uBO. There is no point complaining about it, it's just not for you, it's meant for another kind of users -- you do not have to use it. For the record, it's not for me either (I want/need the full control uBO allows me), but I want to offer an option for those who use uBO as an install-and-forget blocker without ever interacting with it.
 
 The issue is that AdNauseam would require Service Workers to in order to collect the ads, visit them, and "send" them to the AdVault. Perhaps if we want to have AdNauseam for v3 we would have to implement our own service worker not based on uBlock any longer.
 
In order to implement our own Service Worker, we would need to have close attention to the issue raised here:

> Worrying CPU/memory usage due to the fact that the service worker associated with the extension is constantly evicted and re-spawned, every few seconds or when navigating to a webpage (observed from the browser's Task Manager, ShiftEsc) -- thereby defeating [one of stated benefits of MV3](https://developer.chrome.com/docs/extensions/mv3/intro/platform-vision/#where-were-headed) regarding performance.

-------------

Taken this into consideration it seems that our option would be to release a "lite" version of AdNauseam, largely based on UBOLite, in which we would have to make our own brand-new implementation of clicking and collecting, using UBOLite only as a baseline for general adBlocking. 

For other browsers AdNauseam would work as it is, keeping track with uBlock Origin main implementations.

-------------

I am far from having complete understanding of uBlock as well as the new implications of the v3 manifest, __so feel to add ideas to this discussion__. Links to the issues related to v3 manifest will be added bellow.

**Comment by mneunomne** [comment]
see https://github.com/dhowe/AdNauseam/issues/2189

**Comment by venkat2305** [comment]
have you guys started working on mv3 version? only two months left before they stop supporting mv2

**Reply by mneunomne** [reply]
We are working on it. since are not in the chrome web store, we still have a bit more time since it will be only from June 2023 that chrome will completely disable mv2 on stable builds.  

More details on the timeline here: https://developer.chrome.com/docs/extensions/mv3/mv2-sunset/
